src/collectors/ceph/ceph.py

```diff
@@ -35,30 +45,120 @@ def flatten_dictionary(input, sep='.', prefix=None):
 
     produces::
 
-      [('a.b', 10), ('c', 20)]
+      [([a,b], 10), ([c], 20)]
     """
-    for name, value in sorted(input.items()):
-        fullname = sep.join(filter(None, [prefix, name]))
+    if path is None:
+        path = []
+
+    for name, value in sorted(input_dict.items()):
+        path.append(name)
         if isinstance(value, dict):
-            for result in flatten_dictionary(value, sep, fullname):
+            for result in flatten_dictionary(value, path):
                 yield result
         else:
-            yield (fullname, value)
+            yield (path[:], value)
+        del path[-1]
+
+
+def lookup_dict_path(d, path, extra=list()):
+    """Lookup value in dictionary based on path + extra.
+
+    For instance, [a,b,c] -> d[a][b][c]
+    """
+    element = None
+    for component in path + extra:
+        d = d[component]
+        element = d
+    return element
+
+
+class CalledProcessError(Exception):
+    pass
+
+
+def _popen_check_output(*popenargs):
```

Comment:
I think it's reasonable. pypi lists subprocess32 as available for python 2.4-2.7. It would mean adding it to setup.py
https://github.com/python-diamond/Diamond/blob/master/setup.py#L146

Does that sound like the right thing to do?